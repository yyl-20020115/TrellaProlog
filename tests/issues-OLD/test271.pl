a(1).
a(2).
a(_) :- throw(e).

from_generator(Goal, Value, Optional) :-
	catch(Goal, Error, true),
	(	var(Error) ->
		Optional = optional(Value)
	;	Optional = empty,
		!
	).
from_generator(_, _, empty).

:-initialization(main).

main :-
	findall(Optional, from_generator(a(X), X, Optional), Optionals),
	write(Optionals), nl.

