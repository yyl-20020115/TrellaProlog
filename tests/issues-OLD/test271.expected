[optional(1),optional(2),empty]
